lightning-jsonrpc.patch

```diff
@@ -0,0 +1,69 @@
+diff --git a/cli/lightning-cli.c b/cli/lightning-cli.c
+index bb120f1b..7b9c01a5 100644
+--- a/cli/lightning-cli.c
++++ b/cli/lightning-cli.c
+@@ -522,10 +522,12 @@ int main(int argc, char *argv[])
+ 	fd = socket(AF_UNIX, SOCK_STREAM, 0);
+ 	if (strlen(rpc_filename) + 1 > sizeof(addr.sun_path))
+ 		errx(ERROR_USAGE, "rpc filename '%s' too long", rpc_filename);
+-	strcpy(addr.sun_path, rpc_filename);
++	addr.sun_path[0] = '\0';
++	strcpy(addr.sun_path + 1, rpc_filename);
+ 	addr.sun_family = AF_UNIX;
```

Comment:
I don't quite follow this change. `addr.sun_path` now has a string terminator at the beginning?

Reply:
This patch is not related to the building process, but related to use c-lightning in an android device that doesn't support communication using traditional AF_UNIX socket file. The '\0' swap allows to use abstract namespace socket for inter-process communication (also available on linux kernel that holds in its memory space). I think there are some security concert on accessing them.